backend-rust/src/indexer.rs

```diff
@@ -2038,10 +2111,19 @@ impl PreparedBakerEvent {
             BakerEvent::BakerSetOpenStatus {
                 baker_id,
                 open_status,
-            } => PreparedBakerEvent::SetOpenStatus {
-                baker_id:    baker_id.id.index.try_into()?,
-                open_status: open_status.to_owned().into(),
-            },
+            } => {
+                let open_status = open_status.to_owned().into();
+                let move_delegators = if matches!(open_status, BakerPoolOpenStatus::ClosedForAll) {
```

Comment:
Why are we not doing this when setting the BakerPoolOpenStatus to ClosedForAll instead?